test/resources/complex_xml_transformed.yml

```diff
@@ -0,0 +1,626 @@
+        hydronephrosis: ''
+        s-category: ''
+        postoperative_nadirs-category: ''
+    - 0
+  - - ECosdXmlStaging
+    - tcategory_finalpretreatment: ngv1KRgf5WOXuc7
+      ncategory_finalpretreatment: V8h1It3jLl4Izvf
+      mcategory_finalpretreatment: 483wBKjug9MRB0l
+      tnmstagrofinpre: o5aUmT28ffUyKWj
+      org_site_tnm_group_final_pt: LN0ZH
+      stadatfinpresta: '2023-10-26'
+      tcategory_integratedstage: lqvDg6O5jw7QDFE
+      ncategory_integratedstage: kDI6Bnk5qbUN64B
+      mcategory_integratedstage: e0fxG2RfYosEFE5
+      tnmstagegrouping_integrated: oOBlEwwH3GtZKWQ
+      org_site_tnm_group_int: 3KLUG
+      stadatintsta: '2023-10-26'
+      tnm_coding_edition: '1'
+      tnmedition: QL
+      :rawtext:
+        tcategory_finalpretreatment: ngv1KRgf5WOXuc7
+        ncategory_finalpretreatment: V8h1It3jLl4Izvf
+        mcategory_finalpretreatment: 483wBKjug9MRB0l
+        tnmstagegrouping_finalpretreatment: o5aUmT28ffUyKWj
+        org_site_tnm_group_final_pt: LN0ZH
+        tnmstagegroupingdate_finalpretreatment: '2023-10-26'
+        tcategory_integratedstage: lqvDg6O5jw7QDFE
+        ncategory_integratedstage: kDI6Bnk5qbUN64B
+        mcategory_integratedstage: e0fxG2RfYosEFE5
+        tnmstagegrouping_integrated: oOBlEwwH3GtZKWQ
+        org_site_tnm_group_int: 3KLUG
+        tnmstagegroupingdate_integrated: '2023-10-26'
+        tnm_coding_edition: '1'
+        tnmeditionnumber: QL
+        changstagingformedulloblastoma: ''
+        wilmstumourstage: ''
+        intl_risk_group_staging_system_stage: ''
+        pretextstagingsystemstage: ''
+        pretextannotationfactors: ''
+        internationalstagingsystemforretinoblastoma: ''
+        annarborstage: ''
+        binetstage: ''
+        rissstageformyeloma: ''
+        murphy_stjudestage: ''
+        childrens_oncology_group_cog_staging_system_stage: ''
+        central_nervous_system_involvement: ''
+        barcelonacliniclivercancerstagebclcstage: ''
+        stagegrouping_testicular: ''
+        annarborsymptoms: ''
+        annarborextranodality: ''
+        annarborbulk: ''
+        annarborsplenicinvolvement: ''
+    - 0
+  - - ECosdXmlRecurrence#NPP
+    - :rawtext:
+        dateofnonprimarydiagnosis: ''
+        originalprimarydiagnosis_icd: ''
+        metastatictype: ''
+        metastaticsite: ''
+    - 0
+  - - ECosdXmlProgression#NPP
+    - :rawtext:
+        dateofnonprimarydiagnosis: ''
+        cancer_progression_icd: ''
+        metastatictype: ''
+        metastaticsite: ''
+    - 0
+  - - ECosdXmlRecurrence
+    - :rawtext:
+        palliativecarespecialistseenindicator_cancerrecurrence: ''
+        relapse_method_of_detection: ''
+    - 0
+  - - ECosdXmlProgression#1
+    - metastatictype_1: '02'
+      metastaticsite_1: '07'
+      dateofnonprimarydiagnosis: '2023-10-26'
+      :rawtext:
+        metastatictype_1: '02'
+        metastaticsite_1: '07'
+        dateofnonprimarydiagnosis: '2023-10-26'

```

Comment:
Is it expected behaviour to have `metastatictype_1` but no unnumbered `metastatictype` field in the `rawtext` here?

For `address`, we seem to have both. The `ECosdXmlProgression#NPP` entries immediately above have a blank `metastatictype` field (but no indexed entry).

Reply:
Yes, this is expected.

If there are multiple instances of an xpath (there are 2 additional pairs in `ECosdXmlProgression#2`) then the rawtext will be incremented. If the data is not present (e.g `ECosdXmlProgression#NPP`) then the non-incremented key will appear with a blank value.

Similarly, if there is only a single instance of the xpath, the non-incremented key will appear, but with a value.